### ISSUE TITLE
 error reading magic number of file: EOF after OS hard kill

### ISSUE THREAD

**Issue by wosiu** [influxdata/influxdb#9830]
### Bug report

__System info:__ [Include InfluxDB version, operating system name, and other relevant details]
Ubuntu 16.04
influxdb_1.5.1_amd64

__Actual behavior:__ [What actually happened]
Getting 
```
[shard 38] error opening memory map for file /var/lib/influxdb/data/OCR_METRICS_DB/autogen/38/000000001-000000001.tsm: init: error reading magic number of file: EOF
```

__Steps to reproduce:__
1. Kill whole system hard
2. Cannot write to influxDB after system start

__Expected behavior:__ [What you expected to happen]
Move broken files out of data storage dir and start new, or fix broken file automatically.

__Manual workaround__
rm -rf /var/lib/influxdb/data/OCR_METRICS_DB/autogen/38

**Comment by rbetts** [comment]
Curious if https://github.com/influxdata/influxdb/pull/9785 catches the case you see?

**Comment by benbjohnson** [comment]
It sounds like there's a zero length file being generated somehow. We put in a fix to rename, log, & skip corrupt files here: https://github.com/influxdata/influxdb/pull/9785

**Comment by e-dard** [comment]
Closed via #9785.
